keps/sig-apps/20190318-PodDisruptionBudget-graduation-to-stable.md

```diff
@@ -93,6 +93,18 @@ but will still require `DisruptionsAllowed` to be larger than 0 for it to be evi
 arise from this. For example if we have a PDB with `MinAvailable  = 1` and 10 pods that are all in the CrashLoop state 
 (`Running`, but not `Ready`), we will not be allowed to evict any of the pods.
 
+To fix this, we will unify the logic between the disruption controller and the eviction REST handler so that a pod that 
+is not considered ready by the controller will always be allowed to be evicted (even if DisruptionsAllowed is 0). 
+Also note that eviction attempts for pods that are not ready but could transition to being ready in the future (e.g. 
+are not in a terminal state) will be made conditional on the resourceVersion at which their unready state was determined 
+to avoid a race condition between pod readiness and eviction. If the conditional eviction fails with a conflict error, 
+it will be retried internally by the eviction REST handler. There is pretty much agreement that this is safe for pods 
```

Comment:
The language for the "not running" and "running but not ready" aspects must be more assertive and definite before this can be merged. It's currently unclear if merging this means we have decided it is safe, or is not safe.

Reply:
Agree. I just wanted to point out the discussion around it. I hope to get some more feedback on this PR and then update it to reflect the consensus.